[release-1.7] 🌱 Grouped CI related cherry-picks

chrischdi opened the pull request:
This is a manual cherry-pick of:

- #2180
- #2167
- #2172
  - only CI parts
- #2174
- #2211
- #2147
  - only CI parts partially

sbueringer commented:
Changes look good. We probably have to find a way to ignore some markdown for the linkchecker (the stuff we dropped on newer branches)

chrischdi commented:
> Changes look good. We probably have to find a way to ignore some markdown for the linkchecker (the stuff we dropped on newer branches)

I did not find a way to exclude them. The directory `contrib/haproxy/openapi/docs` did not get changed for the last 4 years and were introduced in #705 . 

We don't use them either so instead of a big toil challenge I removed the entire directory.

chrischdi commented:
> Sounds good. Looks like you currently only pushed a revert of the flavorgen change? (on this PR, 1.6 + 1.5 looks fine +/- further linkchecker findings)

Jep, missed pushing this branch too...

sbueringer commented:
/lgtm
/approve
